source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -45,16 +45,32 @@ ShaderInput::ShaderInput(ShaderNode* node, const TypeDesc* type, const string& n
 
 void ShaderInput::makeConnection(ShaderOutput* src)
 {
-    breakConnection();
-    _connection = src;
-    src->_connections.insert(this);
+    // Make sure this is a new connection.
+    if (src != _connection)
+    {
+        // Break the old connection.
+        breakConnection();
+        if (src)
+        {
+            // Make the new connection.
+            _connection = src;
+            src->_connections.push_back(this);
+        }
+    }
 }
 
 void ShaderInput::breakConnection()
 {
     if (_connection)
     {
-        _connection->_connections.erase(this);
+        // Find and erase this input from the connected output's connection vector.
+        ShaderInputVec& connected = _connection->_connections;
+        ShaderInputVec::iterator it = std::find(connected.begin(), connected.end(), this);
```

Comment:
Breaking a connection is now worst case O(n) with regards to the number of inputs connected to. But this number is very small. In the large majority of cases there is just a single input connected downstream.